src/amo/components/AddonReview.js

```diff
@@ -19,40 +17,52 @@ export class AddonReviewBase extends React.Component {
     errorHandler: PropTypes.object.isRequired,
     i18n: PropTypes.object.isRequired,
     review: PropTypes.object.isRequired,
-    router: PropTypes.object.isRequired,
     updateReviewText: PropTypes.func,
   }
 
-  onSubmit = (event) => {
+  constructor(props) {
+    super(props);
+    this.state = { reviewTitle: null };
+    if (props.review) {
+      // TODO: also do this in componentWillReceiveProps.
+      this.state.reviewTitle = props.review.title;
+      this.state.reviewBody = props.review.body;
+    }
+    this.overlayCard = null;
+    this.reviewTextarea = null;
+    this.reviewTitleInput = null;
+  }
+
+  onSubmit = (event, { overlayCard = this.overlayCard } = {}) => {
```

Comment:
Generally, @mstriemer advocated for splitting all our `onX` methods apart into two things: the `onSubmit` type thing that would take the event as the argument and then another method (here, I guess it'd be `updateReview`) that actually had all the logic. It makes it easy to know what to name the event handlers themselves and means multiple handlers can call the same function. It should means testing the logic of the reviewupdate method doesn't need a stubbed out event.

Short version is we've been keeping the event handler and logic in separate methods which is a bit more code but easier to follow in my and @mstriemer's opinion. Could we do that here?